io-engine/src/grpc/v1/pool.rs

```diff
@@ -162,6 +165,124 @@ impl PoolService {
             client_context: std::sync::Arc::new(tokio::sync::RwLock::new(None)),
         }
     }
+
+    async fn serialized<T, F>(
+        &self,
+        ctx: GrpcClientContext,
+        pool_name: Option<String>,
+        global_operation: bool,
+        write_lock: bool,
+        f: F,
+    ) -> Result<T, Status>
+    where
+        T: Send + 'static,
+        F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
+    {
+        async fn acquire_lock(
```

Comment:
can this code be reused across services?

Reply:
Move to a trait. We can have different implementation for each service in future, if needed. Default implementation is kept in trait.